File: src/main/java/fr/yodamad/svn2git/service/util/GitlabAdmin.java
package fr.yodamad.svn2git.service.util;

import fr.yodamad.svn2git.config.ApplicationProperties;
import org.gitlab4j.api.GitLabApi;
import org.gitlab4j.api.GroupApi;
import org.gitlab4j.api.ProjectApi;
import org.gitlab4j.api.UserApi;
import org.springframework.beans.factory.config.ConfigurableBeanFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.context.annotation.ScopedProxyMode;
import org.springframework.stereotype.Component;

@Component
@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE, proxyMode = ScopedProxyMode.TARGET_CLASS)
public class GitlabAdmin {

    /**
     * Gitlab API wrapper.
     */
    private GitLabApi gitLabApi;

    public GitlabAdmin(ApplicationProperties applicationProperties) {
        gitLabApi = new GitLabApi(applicationProperties.gitlab.url, applicationProperties.gitlab.token);
        gitLabApi.setIgnoreCertificateErrors(true);
    }

    // Getter on api

    public GitLabApi api() {
        return this.gitLabApi;
    }

    // Convenience getters

    public GroupApi groupApi() {
        return this.gitLabApi.getGroupApi();
    }

    public UserApi userApi() {
        return this.gitLabApi.getUserApi();
    }

    public ProjectApi projectApi() {
        return this.gitLabApi.getProjectApi();
    }

    // Setter on GitLabApi only

    public void setGitLabApi(GitLabApi gitLabApi) {
        this.gitLabApi = gitLabApi;
    }

}

File: src/main/kotlin/fr/yodamad/svn2git/service/util/GitlabAdmin.kt
package fr.yodamad.svn2git.service.util

import fr.yodamad.svn2git.config.ApplicationProperties
import org.gitlab4j.api.GitLabApi
import org.gitlab4j.api.GroupApi
import org.gitlab4j.api.ProjectApi
import org.gitlab4j.api.UserApi
import org.springframework.beans.factory.config.ConfigurableBeanFactory
import org.springframework.context.annotation.Scope
import org.springframework.context.annotation.ScopedProxyMode
import org.springframework.stereotype.Component
import javax.annotation.PostConstruct

@Component
@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE, proxyMode = ScopedProxyMode.TARGET_CLASS)
open class GitlabAdmin(val applicationProperties: ApplicationProperties) {

    @PostConstruct
    open fun initApi() {
        gitLabApi = GitLabApi(applicationProperties.gitlab.url, applicationProperties.gitlab.token)
        gitLabApi.ignoreCertificateErrors = true
    }

    /**
     * Gitlab API wrapper.
     */
    lateinit var gitLabApi : GitLabApi

    // Getter on api
    open fun api(): GitLabApi {
        return gitLabApi
    }

    // Convenience getters
    open fun groupApi(): GroupApi {
        return gitLabApi.groupApi
    }

    open fun userApi(): UserApi {
        return gitLabApi.userApi
    }

    open fun projectApi(): ProjectApi {
        return gitLabApi.projectApi
    }

    open fun setGitlabApi(api: GitLabApi) {
        gitLabApi = api
    }

    /*init {
        gitLabApi = GitLabApi(applicationProperties.gitlab.url, applicationProperties.gitlab.token)
        gitLabApi.ignoreCertificateErrors = true
    }*/
}
